fdbserver/TagPartitionedLogSystem.actor.cpp

```diff
@@ -184,7 +184,9 @@ struct TagPartitionedLogSystem : ILogSystem, ReferenceCounted<TagPartitionedLogS
 	Optional<Version> recoveredAt;
 	Version knownCommittedVersion;
 	LocalityData locality;
-	std::map< std::pair<UID, Tag>, std::pair<Version, Version> > outstandingPops;  // For each currently running popFromLog actor, (log server #, tag)->popped version
+	std::map<std::pair<UID, Tag>, std::pair<Version, Version>> outstandingPops;
+	// For each currently running popFromLog actor, (logID, tag)->(max popped version, durableKnownCommittedVersion)
+	// Q: why do we need tag and durableKnownCommittedVersion?
```

Comment:
@sfc-gh-almiller 
The second version is "durableKnownCommittedVersion". 
I somehow remembered it is there for recovery. But I still don't know why. 
Can you please explain it a bit?

Reply:
Overall, knownCommittedVersion gives the lower bound of what data will need to be copied into the next generation to restore the replication factor.

It's not immediately clear to me why it'd be paired with something for popping.  I'm guessing it's because of something like that it serves as a minimum version of what data would be on a TLog in the next generation, and sending a pop for anything less than that would be somewhat absurd?  I'd need to go dig into the code for this specific use.